### PR TITLE
dev-java/eclipse-ecj: handle dependency on eselect-java - W I P

### PR DESCRIPTION
**Pull request by vaukai**
https://qa-reports.gentoo.org/output/eapi-per-eclass/java-pkg-2.eclass/7.txt

### Related pull requests:

- https://github.com/gentoo/gentoo/pull/38469
- https://github.com/gentoo/gentoo/pull/32051
- https://github.com/gentoo/gentoo/pull/39589

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by fordfrog** [comment]
some stuff merged now

**Comment by fordfrog** [comment]
merged some stuff

**Comment by vaukai** [comment]
> merged some stuff

for [java-pkg-simple.eclass](https://qa-reports.gentoo.org/output/eapi-per-eclass/java-pkg-simple.eclass/7.txt) it is done. now pending for stabilization.

**Comment by fordfrog** [comment]
merged the drop

**Comment by vaukai** [comment]
> merged the drop

and rebased, no further changes.

**Comment by fordfrog** [comment]
merged some stuff

**Comment by vaukai** [comment]
> merged some stuff

rebased and removal of old `protobuf-java` added.

File: dev-java/eclipse-ecj/eclipse-ecj-4.33.ebuild [hunk 1]
```diff
@@ -22,16 +22,16 @@ SLOT="4.33"
 KEYWORDS="~amd64 ~arm64 ~ppc64"
 
 BDEPEND="app-arch/unzip"
-COMMON_DEP="app-eselect/eselect-java"
```
Comment (fordfrog) [inline dev-java/eclipse-ecj/eclipse-ecj-4.33.ebuild:25]: the dependency is not useless as we have `eselect ecj`:
```
$ eselect ecj
Usage: eselect ecj <action> <options>

Standard actions:
  help                      Display help text
  usage                     Display usage information
  version                   Display version information

Extra actions:
  list                      List available ECJ targets
  set <target>              Set a new ECJ target
    target                    Target name or number (from 'list' action)
  show                      Show current ECJ target
  update <target>           Set the ECJ target to the latest if the current target is invalid or if the given target is the latest
    target (optional)         Target name (from 'list' action)
```

**Comment by fordfrog** [comment]
everything merged except the one that i commented on today.

File: dev-java/eclipse-ecj/eclipse-ecj-4.33.ebuild (hunk 1 above)
Comment (vaukai) [reply dev-java/eclipse-ecj/eclipse-ecj-4.33.ebuild:25]: Should I put `optfeature` (or what was its name?)?

Comment (fordfrog) [reply dev-java/eclipse-ecj/eclipse-ecj-4.33.ebuild:25]: i'd simply keep the dependency, it just can be moved to `RDEPEND`.

Comment (vaukai) [reply dev-java/eclipse-ecj/eclipse-ecj-4.33.ebuild:25]: > i'd simply keep the dependency, it just can be moved to `RDEPEND`.

But it is definitely not needed, just optional, for running `eclipse-ecj`. Isn't this a typical use case for `optfeature`?

Comment (fordfrog) [reply dev-java/eclipse-ecj/eclipse-ecj-4.33.ebuild:25]: well, it's the same case like for `maven-bin`. you use `eselect` to pick the version that is used from command line. `maven-bin` ebuild just goes further and also updates the selected implementation which we don't do for `ecj`, though we might or maybe even should.

Comment (vaukai) [reply dev-java/eclipse-ecj/eclipse-ecj-4.33.ebuild:25]: > well, it's the same case like for `maven-bin`. you use `eselect` to pick the version that is used from command line. `maven-bin` ebuild just goes further and also updates the selected implementation which we don't do for `ecj`, though we might or maybe even should.

@fordfrog I will close this PR leaving the `eclipse-ecj-4.33` ebuild as is. It really doesn't hurt.

**Pull request closed without merging**